connect/runtime/src/main/java/org/apache/kafka/connect/storage/FileOffsetBackingStore.java

```diff
@@ -78,6 +85,27 @@ private void load() {
                 ByteBuffer key = (mapEntry.getKey() != null) ? ByteBuffer.wrap(mapEntry.getKey()) : null;
                 ByteBuffer value = (mapEntry.getValue() != null) ? ByteBuffer.wrap(mapEntry.getValue()) : null;
                 data.put(key, value);
+                if (key != null) {
+                    try {
+                        // The topic parameter is irrelevant for the JsonConverter which is the internal converter used by
+                        // Connect workers.
+                        List<Object> keyValue = (List<Object>) keyConverter.toConnectData("", key.array()).value();
+                        // The key should always be of the form [connectorName, partition] where connectorName is a
+                        // string value and partition is a Map<String, Object>
+                        String connectorName = (String) keyValue.get(0);
+                        Map<String, Object> partition = (Map<String, Object>) keyValue.get(1);
+                        if (!connectorPartitions.containsKey(connectorName)) {
+                            connectorPartitions.put(connectorName, new HashSet<>());
+                        }
```

Comment:
Believe this can be one line:
```suggestion
                connectorPartitions.computeIfAbsent(connectorName, ignored -> new HashSet<>());
```